website/common/script/ops/feed.js

```diff
@@ -21,34 +19,25 @@ function evolve (user, pet, req) {
     egg: pet.text(req.language),
   }, req.language);
 }
-    function handleMountAchievements(user) {
-        //Mount collector
-        console.log(0);
-        console.log(user);
-        console.log(user.achievements.mountMaster || 0);
-        var retrospectiveCount = (user.achievements.mountMaster || 0) * 90 + (Object.keys(user.items.mounts).length || 0);
-        console.log("retro" + retrospectiveCount);
-        console.log(Object.keys(user.items.mounts).length);
-        user.achievements.mountCollector = Math.max(retrospectiveCount, user.achievements.mountCollector || 0);
-        console.log(user.achievements.mountCollector);
-        switch (user.achievements.mountCollector) {
-            case 1:
-                Achievement.displayAchievement('mountCollector1');
-                break;
-            case 10:
-                Achievement.displayAchievement('mountCollector10');
-                break;
-            case 25:
-                Achievement.displayAchievement('mountCollector25');
-                break;
-            case 50:
-                Achievement.displayAchievement('mountCollector50');
-                break;
-        }
-
+function handleMountAchievements(_user) {
+    console.log("MC-1 " + _user.achievements.mountCollector);
 
+    //Mount collector
+    var mountCount = 0;
+    for (var i in _user.items.mounts) {
+        if (_user.items.mounts[i]) mountCount++;
     }
-module.exports = function feed (user, req = {}) {
+    //tracks mounts raised on mobile, or before the achievement was added
+    console.log("MC0 " + _user.achievements.mountCollector);
+    console.log(mountCount);
+    //var retrospectiveCount = (user.achievements.mountMasterCount || 0) * 90 + (mountsCount || 0);
+    //console.log(retrospectiveCount);
+    console.log("MC1 " + _user.achievements.mountCollector);
+    _user.achievements.mountCollector = Math.max(mountCount, (_user.achievements.mountCollector || 0) + 1);
```

Comment:
For some reason, user.achievements.mountCollector sometimes jumps  +91 either **between** calls to evolve, or after restarting the server. It seems to have something to do with line 36, as if it is Math.max(0, (...) + 1) it doesn't do it (at least not between function calls, I am not sure about the server reset yet.